### DISCUSSION TITLE
Cursor style over SSH in vi mode

### DISCUSSION THREAD

**Discussion by NAlexPear** [ghostty-org/ghostty#3797] [Q&A] [answered]
Thanks for the work that's gone into `ghostty` already, and looking forward to its progress after 1.0!

I've been testing out `ghostty` as a replacement for `kitty`, and the last sticking point is in `ssh`: cursor style is not respected over an `ssh` connection when switching modes in `vi` mode. To put it more clearly:

0. I've copied `TERMINFO` from the host to the child as-documented, confirming that there's an `xter-ghostty` entry on both host and child `infocmp`.
1. I'm using the default cursor configuration on the host machine (Linux), with vi mode enabled in my `.zshrc` using `bindkey -v`
2. I'm using `ssh` to connect to a macOS machine on the local network, with the same vi mode enabled via `.zshrc` using `bindkey -v`
3. When I am in insert mode on the host, the cursor is a blinking bar. When I am `ssh`'d into the macbook in insert mode, the cursor is a blinking block.

Is this :point_up: a bug with `TERMINFO`, or a misunderstanding on my part? When using the `ssh` kitten with `kitty`, the cursor behaviors are sync'd correctly.

**Comment by dgl** [comment, marked as answer]
The cursor style is implemented by the shell integration. You'll need to manually configure it on the remote system: https://ghostty.org/docs/features/shell-integration#manual-shell-integration-setup

**Reply by NAlexPear** [reply]
Thanks @dgl, I was missing the fact the shell integration handled that terminal style and that `GHOSTTY_RESOURCES_DIR` wasn't being propagated over `ssh`.

For anyone else with a similar setup/issue, the way that I handled this specific case (Linx -> macOS, all over a local network):

1. On the macOS machine, [adding the explicit `zsh` integration snippet](https://github.com/ghostty-org/ghostty/blob/main/src/shell-integration/README.md#zsh) to the top of my `.zshrc` file
2. In my host Linux machine's `~/.ssh/config` file, adding an entry like:

    ```bot
    Host 123.456.7.890 # my macbook's IP on the local network
      User my-local-user
      SetEnv GHOSTTY_RESOURCES_DIR=/Applications/Ghostty.app/Contents/Resources/ghostty
    ```
    This :point_up: sets the resources dir to the same value as what would normally exist on the macOS machine.
3. In my macbook's `/etc/ssh/sshd_config.d/100-macos.conf` file, I modified the `AcceptEnv` line to accept `GHOSTTY_RESOURCES_DIR` from incoming connections. The final entry after modification looks like:

    ```bot
    AcceptEnv LANG LC_* GHOSTTY_RESOURCES_DIR
    ```

And now when I open up an SSH connection to my macbook, the shell integration is properly sourced.

This works for me, but this flow could really use some love in the future. I'd love to see shell integration propagate over `ssh`, specifically, even if that shell integration was through a separate `kitten`-style module instead of `ssh` directly!
